edgelet/edgelet-docker/src/image_prune_data.rs

```diff
@@ -0,0 +1,646 @@
+            time,
+        );
+        image_map.insert(
+            "sha256:553fd62d98efd413c6c97ded7e6c6c46fc38c7b1e30a2bffd7e08c04f0d65863".to_string(),
+            time,
+        );
+        image_map.insert(
+            "sha256:79386db4871013d571ced41443e6384ccc82ccd7f4988ecec4d5f91cbd488a99".to_string(),
+            time,
+        );
+
+        let _write = write_images_with_timestamp(
+            &image_map,
+            test_file_dir
+                .join(TMP_FILENAME)
+                .into_os_string()
+                .into_string()
+                .unwrap(),
+            test_file_dir
+                .join(IMAGE_USE_FILENAME)
+                .into_os_string()
+                .into_string()
+                .unwrap(),
+        );
+
+        let curr_time: String = format!("{}{}", Utc::now().hour(), Utc::now().minute());
+        let mut settings = ImagePruneSettings::new(
+            Duration::from_secs(30),
+            Duration::from_secs(5),
+            curr_time,
+            false,
+        );
+        let mut image_use_data = ImagePruneData::new(&test_file_dir, Some(settings)).unwrap();
+
+        let mut in_use_image_ids: HashSet<String> = HashSet::new();
+        in_use_image_ids.insert(
+            "sha256:670dcc86b69df89a9d5a9e1a7ae5b8f67619c1c74e19de8a35f57d6c06505fd4".to_string(),
+        );
+        in_use_image_ids.insert(
+            "sha256:7a45202c8491b92b7e7a9a0cbf887079fcdb86ea3cb0b7a4cb8f5491281e985d".to_string(),
+        );
+        in_use_image_ids.insert(
+            "sha256:a40d3130a63918663f6e412178d2e83010994bb5a6bdb9ba314ca43013c05331".to_string(),
+        );
+        in_use_image_ids.insert(
+            "sha256:85fdb1e9675c837c18b75f103be6f156587d1058eced1fc508cdb84a722e4f82".to_string(),
+        );
+
+        unsafe { sleep(6) };
+
+        // image prune disable... don't remove stuff
+        let mut images_to_delete = image_use_data
+            .prune_images_from_file(in_use_image_ids.clone())
+            .unwrap();
+        assert!(images_to_delete.is_empty());
+
+        // image prune enabled, remove stuff
+        let curr_time: String = format!("{}{}", Utc::now().hour(), Utc::now().minute());
+        settings = ImagePruneSettings::new(
+            Duration::from_secs(30),
+            Duration::from_secs(5),
+            curr_time,
+            true,
+        );
+        image_use_data = ImagePruneData::new(&test_file_dir, Some(settings)).unwrap();
+
+        images_to_delete = image_use_data
+            .prune_images_from_file(in_use_image_ids)
+            .unwrap();
+        assert!(images_to_delete.len() == 2);
+
+        // cleanup
+        std::fs::remove_dir_all(test_file_dir).unwrap();
+    }
+
+    /* =============================================================== MORE TESTS ============================================================ */
+
+    #[test]
+    #[serial]
+    #[should_panic]
```

Comment:
Where does this panic and why? A comment would be helpful.

Reply:
please see comment below.